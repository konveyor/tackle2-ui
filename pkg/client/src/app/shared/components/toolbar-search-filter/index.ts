export { ToolbarSearchFilter } from "./toolbar-search-filter";
