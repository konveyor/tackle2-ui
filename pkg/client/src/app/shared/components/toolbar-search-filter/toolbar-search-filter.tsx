import React, { useState } from "react";

import { InputGroup } from "@patternfly/react-core";
import { SimpleFilterDropdown } from "@app/shared/components";

interface FilterOption {
  key: string;
  name: React.ReactNode;
  input: React.ReactNode;
}

export interface IToolbarSearchFilterProps {
  filters: FilterOption[];
}

export const ToolbarSearchFilter: React.FC<IToolbarSearchFilterProps> = ({
  filters,
}) => {
  const [selectedKey, setSelectedKey] = useState<string | undefined>(
    filters.length > 0 ? filters[0].key : undefined
  );

  const selectedFilter = filters.find((f) => f.key === selectedKey);

  return (
    <InputGroup>
      <SimpleFilterDropdown
        label={selectedFilter?.name}
        options={filters.map((f) => ({ key: f.key, name: f.name }))}
        onSelect={(dropdown) => setSelectedKey(dropdown.key)}
      />
      {selectedFilter?.input}
    </InputGroup>
  );
};
