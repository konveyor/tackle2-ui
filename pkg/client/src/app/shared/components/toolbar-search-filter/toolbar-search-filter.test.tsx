import React from "react";
import { shallow, mount } from "enzyme";
import { ToolbarSearchFilter } from "./toolbar-search-filter";

describe("ToolbarSearchFilter", () => {
  it("Renders without crashing", () => {
    const wrapper = shallow(
      <ToolbarSearchFilter
        filters={[
          {
            key: "filter1",
            name: "Filter 1",
            input: <input cy-data="filter1" />,
          },
          {
            key: "filter2",
            name: "Filter 2",
            input: <input cy-data="filter2" />,
          },
          {
            key: "filter3",
            name: "Filter 3",
            input: <input cy-data="filter3" />,
          },
        ]}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it("Renders without crashing when no filter is provided", () => {
    const wrapper = shallow(<ToolbarSearchFilter filters={[]} />);
    expect(wrapper).toMatchSnapshot();
  });

  it.skip("Select appropiate filter", () => {
    const wrapper = mount(
      <ToolbarSearchFilter
        filters={[
          {
            key: "filter1",
            name: "Filter 1",
            input: <input cy-data="filter1" />,
          },
          {
            key: "filter2",
            name: "Filter 2",
            input: <input cy-data="filter2" />,
          },
          {
            key: "filter3",
            name: "Filter 3",
            input: <input cy-data="filter3" />,
          },
        ]}
      />
    );

    // Open dropdown
    wrapper.find(".pf-c-dropdown__toggle").simulate("click");
    wrapper.update();

    // Select option
    wrapper.find(".pf-c-dropdown__menu-item").at(2).simulate("click");
    expect(wrapper.find("input[cy-data='filter3']")).not.toBeNull();
  });
});
