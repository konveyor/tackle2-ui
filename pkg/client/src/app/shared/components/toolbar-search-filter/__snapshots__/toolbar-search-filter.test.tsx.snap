// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToolbarSearchFilter Renders without crashing 1`] = `
<InputGroup>
  <SimpleFilterDropdown
    label="Filter 1"
    onSelect={[Function]}
    options={
      Array [
        Object {
          "key": "filter1",
          "name": "Filter 1",
        },
        Object {
          "key": "filter2",
          "name": "Filter 2",
        },
        Object {
          "key": "filter3",
          "name": "Filter 3",
        },
      ]
    }
  />
  <input
    cy-data="filter1"
  />
</InputGroup>
`;

exports[`ToolbarSearchFilter Renders without crashing when no filter is provided 1`] = `
<InputGroup>
  <SimpleFilterDropdown
    onSelect={[Function]}
    options={Array []}
  />
</InputGroup>
`;
